lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -166,15 +166,15 @@ tokenize(("<<<<<<<" ++ _) = Original, Line, 1, _Scope, Tokens) ->
 
 tokenize([$0, $x, H | T], Line, Column, Scope, Tokens) when ?is_hex(H) ->
   {Rest, Number, Length} = tokenize_hex(T, [H], 1),
-  tokenize(Rest, Line, Column + 2 + Length, Scope, [{number, {Line, Column, Column + 2 + Length}, Number} | Tokens]);
+  tokenize(Rest, Line, Column + 2 + Length, Scope, [{number, {Line, Column, Column + 2 + Length}, Number, 16} | Tokens]);
```

Comment:
We could instead represent number tokens as

```erlang
{number, {Line, ColumnStart, ColumnEnd}, {Number, Format}}
```

where `Format` is one of `decimal | octal | hex | binary | char`. Notice this allows to get chars (`?x`) formatted well for free. We can then include the `format: :decimal | ...` in the metadata. Wdyt @uohzxela?